components/treelist.tsx

```diff
@@ -256,6 +359,43 @@ function TreeList({
   );
 }
 
+interface TreeListActionProps {
+  isSelected?: boolean;
+  isActive?: boolean;
+  onClick?: () => void;
+  style?: CSSProp;
+  icon?: RemixiconComponentType;
+  caption?: string;
+}
+
+function TreeListAction({
+  isSelected,
+  isActive,
+  onClick,
+  caption,
+  icon: Icon,
+  style,
+}: TreeListActionProps) {
+  return (
+    <ActionItem
+      $isActive={isActive}
+      $isSelected={isSelected}
+      role="button"
+      tabIndex={0}
+      aria-label="tree-list-action"
+      onClick={() => {
+        if (onClick) {
```

Comment:
move it to 379 like:

```js
if (!onClick) onClick = () => {}
// snip snip code
<ActionItem ... onClick={() => onClick()} />
```

reasoning: the `if` is only evaluated once outside, not at each onClick, which although isn't amazing optimization, is a computational optimization nevertheless